src/fides/api/ops/schemas/saas/saas_config.py

```diff
@@ -345,17 +365,38 @@ def _process_param_values(
             if param.references:
                 references = []
                 for reference in param.references:
-                    first, *rest = reference.field.split(".")
+                    resolved_reference = self.resolve_param_reference(
+                        reference, secrets
+                    )
+                    first, *rest = resolved_reference.field.split(".")
                     references.append(
                         (
-                            FieldAddress(reference.dataset, first, *rest),
-                            reference.direction,
+                            FieldAddress(resolved_reference.dataset, first, *rest),
+                            resolved_reference.direction,
                         )
                     )
                 fields.append(ScalarField(name=param.name, references=references))
             if param.identity:
                 fields.append(ScalarField(name=param.name, identity=param.identity))
 
+    @staticmethod
+    def resolve_param_reference(
+        reference: Union[str, FidesopsDatasetReference], secrets: Dict[str, Any]
+    ) -> FidesopsDatasetReference:
+        """
+        If needed, resolves the given `reference` using the provided `secrets` `dict`.
+        For ease of use, the given `reference` can either be a `str` or `FidesopsDatasetReference`,
+        since a `ParamValue`'s `reference` may be of either type.
+
+        If the `reference` is a `str`, then it's used as a key look up a value in the provided secrets dict,
+        and a `FidesopsDatasetReference` is created and returned from the retrieved secrets object.
+
+        If the `reference` is a `FidesopsDatasetReference`, then it's just returned as-is.
+        """
+        if isinstance(reference, str):
+            reference = FidesopsDatasetReference.parse_obj(secrets[reference])
```

Comment:
This seems like it still has that issue where the reference may not exist in `secrets`.   Just pointed this out because if it gets hit when we're building the graph for the traversa, KeyErrors are hard to debug.  I'd like to see a very specific failure if the reference doesn't exist in secrets.
https://github.com/ethyca/fides/pull/1269/files/b9a1c9a5f53597ae50e8f89748ae3e2b6e4e1235#diff-fb8ec5e3fd765b301329795f257fabe2d8763f22822f7876480519fe8598b4b9

Reply:
whoops - missed this one. will address now!

Reply:
ok - should be addressed by https://github.com/ethyca/fides/pull/1269/commits/de0c1a93ad5761938008ba815304721b5fd1b200

Reply:
beautiful thank you